azure_functions_worker/bindings/shared_memory_data_transfer/file_accessor_unix.py

```diff
@@ -0,0 +1,183 @@
+        # From the list of configured directories where memory maps can be
+        # stored, get the list of directories which are valid (either existed
+        # already or have been created successfully for use).
+        self.valid_dirs = self._get_valid_mem_map_dirs()
+
+    def open_mem_map(
+            self,
+            mem_map_name: str,
+            mem_map_size: int,
+            access: int = mmap.ACCESS_READ) -> Optional[mmap.mmap]:
+        """
+        Note: mem_map_size = 0 means open the entire mmap.
+        """
+        if mem_map_name is None or mem_map_name == '':
+            raise SharedMemoryException(
+                f'Cannot open memory map. Invalid name {mem_map_name}')
+        if mem_map_size < 0:
+            raise SharedMemoryException(
+                f'Cannot open memory map. Invalid size {mem_map_size}')
+        fd = self._open_mem_map_file(mem_map_name)
+        if fd is None:
+            logger.warning(f'Cannot open file: {mem_map_name}')
+            return None
+        mem_map = mmap.mmap(fd.fileno(), mem_map_size, access=access)
+        return mem_map
+
+    def create_mem_map(self, mem_map_name: str, mem_map_size: int) \
+            -> Optional[mmap.mmap]:
+        if mem_map_name is None or mem_map_name == '':
+            raise SharedMemoryException(
+                f'Cannot create memory map. Invalid name {mem_map_name}')
+        if mem_map_size <= 0:
+            raise SharedMemoryException(
+                f'Cannot create memory map. Invalid size {mem_map_size}')
+        file = self._create_mem_map_file(mem_map_name, mem_map_size)
+        if file is None:
+            logger.warning(f'Cannot create file: {mem_map_name}')
+            return None
+        mem_map = mmap.mmap(file.fileno(), mem_map_size, mmap.MAP_SHARED,
+                            mmap.PROT_WRITE)
+        if self._is_mem_map_initialized(mem_map):
+            raise SharedMemoryException(f'Memory map {mem_map_name} '
+                                        'already exists')
+        self._set_mem_map_initialized(mem_map)
+        return mem_map
+
+    def delete_mem_map(self, mem_map_name: str, mem_map: mmap.mmap) -> bool:
+        if mem_map_name is None or mem_map_name == '':
+            raise Exception(
+                f'Cannot delete memory map. Invalid name {mem_map_name}')
+        try:
+            fd = self._open_mem_map_file(mem_map_name)
+            os.remove(fd.name)
+        except Exception as e:
+            # In this case, we don't want to fail right away but log that
+            # deletion was unsuccessful.
+            # These logs can help identify if we may be leaking memory and not
+            # cleaning up the created memory maps.
+            logger.error(f'Cannot delete memory map {mem_map_name} - {e}',
+                         exc_info=True)
+            return False
+        mem_map.close()
+        return True
+
+    def _get_allowed_mem_map_dirs(self) -> List[str]:
```

Comment:
@goiri

Reply:
For now, we can't add any other directories to this (e.g. `/tmp`) since the host is only looking in `/dev/shm` and nothing else. First I'll make the change of reading from the env variable in the host and then make changes here (if we think we need to) to add `/tmp`. But that can be a separate PR after the host changes.